src/homework-14/homework-14.ts

```diff
@@ -0,0 +1,193 @@
+const { v4: uuidv4 } = require('uuid');
+
+enum noteType {
+  default = 'default',
+  confirmation = 'confirmation',
+}
+
+interface IEditNote {
+  name?: string;
+  content?: string;
+}
+
+interface ICreateNote {
+  name: string;
+  content: string;
+  type: noteType;
+}
+
+interface INote extends ICreateNote {
+  id: string;
+  dateCreate: Date;
+  dateModify: Date;
+  done: boolean;
+}
+
+// У списку нотаток повинні бути методи для додавання нового запису, видалення, редагування та отримання повної інформації про нотатку за ідентифікаторо
+
+interface ITodoList {
+  notes: INote[];
+  addNote(note: ICreateNote): string;
+  deleteNote(id: string): boolean;
+  info(id: string): string | undefined;
+  edit(id: string, editNote: IEditNote): boolean;
+  makeDone(id: string): boolean;
+  total(): number;
+}
+
+class TodoList implements ITodoList {
+  protected _notes: Partial<Record<string, INote>> = {};
+
+  public get notes(): INote[] {
+    const result: INote[] = [];
+    for (const noteId in this._notes) {
+      const note = this._notes[noteId] as INote;
+      result.push({ ...note });
+    }
+    return result;
+  }
+
+  addNote(note: ICreateNote): string {
+    if (!note.name) throw new Error(`cannot create note with empty name`);
+    if (!note.content) throw new Error(`cannot create note with empty content`);
+    const dateCreate = new Date();
+    const newNote: INote = {
+      id: uuidv4(),
+      ...note,
+      dateCreate: dateCreate,
+      dateModify: dateCreate,
+      done: false,
+    };
+    this._notes[newNote.id] = newNote;
+    return newNote.id;
+  }
+
+  deleteNote(id: string): boolean {
+    if (this._notes[id]) {
+      delete this._notes[id];
+      return true;
+    }
+    return false;
+  }
+
+  info(id: string): string | undefined {
+    if (this._notes[id]) {
+      const note: INote = this._notes[id] as INote;
+      return `id: ${note.id}
+name: ${note.name}
+content: ${note.content}
+date create: ${note.dateCreate.toISOString()}
+date modify: ${note.dateModify?.toISOString()}
+done: ${note.done}`;
+    }
+    return undefined;
+  }
+
+  edit(id: string, editPropertise: IEditNote): boolean {
```

Comment:
Requirements: "There are two types of notes. Default and those that require confirmation when editing." I can't find these requirements. Could you please double-check it?